### PR TITLE
Add override annotations for HadoopTableOperations

### PR DESCRIPTION
**Pull request by lys0716**
Add override annotations for HadoopTableOperations

Signed-off-by: Yishuang Lu <luystu@gmail.com>

### CODE REVIEW

**Comment by jzhuge** [comment]
Running IntelliJ inspection `Missing \@Override annotation` revealed 35 such errors in 11 production files and 10 test files, including the 2 places you are fixing here. Should we fix them all?

**Comment by lys0716** [comment]
> Nice find!
> Running IntelliJ inspection `Missing \@Override annotation` revealed 35 such errors in 11 production files and 10 test files, including the 2 places you are fixing here. Should we fix them all?

**Comment by lys0716** [comment]
Hi John, I just updated the pr. Please let me know if you are seeing more place to add.

File: api/src/main/java/org/apache/iceberg/types/Type.java [hunk 1]
```diff
@@ -112,10 +114,11 @@ Object writeReplace() throws ObjectStreamException {
   }
 
   abstract class NestedType implements Type {
+    @Override
     public boolean isNestedType() {
       return true;
     }
-
+    @Override
```
Comment (rdblue) [inline api/src/main/java/org/apache/iceberg/types/Type.java]: There should be an empty line between the two methods.

Comment (lys0716) [reply api/src/main/java/org/apache/iceberg/types/Type.java]: Thanks! Updated.

**Comment by rdblue** [comment]
Merged. Thanks for contributing this, @lys0716!

**Pull request merged**